portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayIcons.testcase

```diff
@@ -0,0 +1,127 @@
+@component-name = "portal-frontend-infrastructure"
+definition {
+
+	property portal.release = "true";
+	property portal.upstream = "true";
+	property testray.component.names = "Clay";
+	property testray.main.component.name = "User Interface";
+
+	setUp {
+		TestCase.setUpPortalInstance();
+
+		User.firstLoginPG();
+
+		task ("Add a site") {
+			JSONGroup.addGroup(groupName = "Test Site Name");
+		}
+
+		task ("Add a widget page") {
+			JSONLayout.addPublicLayout(
+				groupName = "Test Site Name",
+				layoutName = "Language Test Page Name");
+		}
+
+		task ("Add a Language Selector to page") {
+			JSONLayout.addWidgetToPublicLayout(
+				groupName = "Test Site Name",
+				layoutName = "Language Test Page Name",
+				widgetName = "Language Selector");
+		}
+	}
+
+	tearDown {
+		var testPortalInstance = PropsUtil.get("test.portal.instance");
+
+		if ("${testPortalInstance}" == "true") {
+			PortalInstances.tearDownCP();
+		}
+		else {
+			JSONGroup.deleteGroupByName(groupName = "Test Site Name");
+		}
+	}
+
+	@description = "LPS-152271 Verify that flag icons are present when clicking on localization button"
+	@priority = "5"
+	test FlagIconsCanRender {
+		task ("Given language portlet") {
+			Navigator.gotoSitePage(
+				pageName = "Language Test Page Name",
+				siteName = "Test Site Name");
+		}
+
+		task ("When click on localization button") {
+			AssertElementPresent(
+				key_flagID = "en-us",
+				locator1 = "Icon#FLAG_ID");
```

Comment:
Remove assertions from When steps. Only assertions should be found in Then steps.

Reply:
Understood, thanks.
I did it as a way to verify that the language selector widget had the en-US flag.

Reply:
One way to verify is to use VerifyElementPresent function or WaitForElementPresent.

Another way is to do an if else to help ensure if it's not en-US then change it to en-US.

Reply:
LanguagePortlet.selectLanguagePG pretty much includes the if else

Reply:
Noted, thanks!